deltachat-rpc-server/Cargo.toml

```diff
@@ -22,6 +22,8 @@ tokio = { workspace = true, features = ["io-std"] }
 tokio-util = { workspace = true }
 tracing-subscriber = { workspace = true, features = ["env-filter"] }
 yerpc = { workspace = true, features = ["anyhow_expose", "openrpc"] }
+console-subscriber = "0.4.0"
+tracing = "0.1.40"
```

Comment:
> This will increase the app size for all users of Delta Chat, I don't think we should do this.

so rust has no tree-shaking / include only what is used?

Reply:
So I tested it:
```
-rwxr-xr-x  1 bb  staff  16137432 Oct 23 13:56 target/release/deltachat-rpc-server
-rwxr-xr-x  1 bb  staff  16386520 Oct 23 13:49 target/release/deltachat-rpc-server-before
```
so statement looks incorrect to me.

> This will increase the app size for all users of Delta Chat

only bots and desktop use deltachat-rpc-server

Reply:
After looking into this some more: _Probably_ the unused crates are optimized out, but it's complicated and _probably_ they unnecessarily increase the compile times for every build. I.e. it would still be nice to only pull them in only when needed.

LTO is enabled in our workspace-root-Cargo.toml, so that there is another optimizing step after linking which should remove the unused code. To be sure, one would have to measure, which I could do, but: Compiling the crates, linking them, and then optimizing them out again is some work that will increase compile times. So, I would say that independently of whether the binary size increases, it's worth it to make the new dependencies optional.

Reply:
> probably they unnecessarily increase the compile times for every build

not all dependencies are compiled on each build again, you don't start from scratch every time normally.

But as long as the feature is **only** for `deltachat-rpc-server`, then I won't be against it.
But I also don't see the big benefit, so feel free to make a commit with this change yourself. 🤷

Reply:
> So I tested it: [...]

Wait, so, does this mean that this PR here significantly _decreases_ the binary size?

If so, I guess it's fine, even though I don't understand it, but optimizations are weird - and, contrary to what I thought in the beginning, this change only affects Desktop, not Android & iOS, so, it's not as important as I thought.

Reply:
> Wait, so, does this mean that this PR here significantly decreases the binary size?

I did it two times to be sure, yes that were my results before rebasing, haven't tested again after that.